src/components/TransactionTable/TransactionTable.js

```diff
@@ -0,0 +1,214 @@
+import React, { PureComponent } from 'react';
+import propTypes from 'prop-types';
+import moment from 'moment';
+import { Table } from '../index';
+import { connectTheme } from '../../utils';
+
+import TransactionView from './TransactionView';
+
+const constants = {
+  //DATE_FORMAT: 'YYYY-MM-DD hh:mm a',
+  DATE_FORMAT: 'MM/DD/YY hh:mm a',
+  SEND: 'Sent',
+  RECEIVE: 'Received',
+  COINBASE_RECEIVE: 'Coinbase',
+  MULTIPLE_OUTPUT_COPY: 'Multiple',
+  UNKNOWN_ADDRESS_COPY: 'Unknown',
+};
+
+// what does a list of txns look like?
+// cache by serializing and hashing the list and then doing a comparison?
+// that operation in itself has a large overhead
+
+class TransactionTable extends PureComponent {
+  constructor() {
+    super();
+    this.headers = [
+      'Date',
+      'Send/Receive',
+      'Wallet',
+      'Amount',
+      'Account',
+      'Confirmations',
+      'Recipient',
+    ];
+    this.expandHeight = 250;
+
+    // keep state of txid to transaction info?
+  }
+
+  static get propTypes() {
+    return {
+      transactions: propTypes.array,
+    };
+  }
+  static get defaultProps() {
+    return {
+      // list of transactions from bcoin api
+      transactions: [],
+    };
+  }
+
+  // differentiate between ingoing and outgoing
+  // based on path dictionary in inputs or outputs
+  // in transaction object
+
+  // bcoin api
+  // deposits:
+  // null path in the inputs
+  // output with path defined and change false
+  //
+  // withdrawals:
+  // input with defined path and change false
+  // output with path defined and change true
+  //
+
+  // also build expandedData?
+  formatTableData(data, wallet) {
```

Comment:
This was the functionality you were talking about putting into a utility right?

Reply:
To be honest, this is where most of the logic of the component is. In which case, it might be the cleanest to offer this as a utility and then you use this to just create the props that are passed directly to the normal `Table` component. This is more or less what I did with some chain and block utilities in bpanel-utils for the RecentBlocks table.